### PR TITLE
update filter and add logging tests #ROBO-2954

### PR DESCRIPTION
**Pull request by mihainradu**

### CODE REVIEW

File: UiPath.FreeRdpClient/UiPath.FreeRdpClient.Tests/TestInfra/TestHost.cs [hunk 1]
```diff
@@ -73,25 +77,31 @@ private IHost CreateHost()
...
             }
 
             if (_messageSink != null)
-                l.AddXUnit(_messageSink, configure);
+            {
+                loggingBuilder.AddXUnit(_messageSink, ConfigureXUnit);
+            }
 
-            l.SetMinimumLevel(LogLevel.Trace);
+            loggingBuilder.SetMinimumLevel(LogLevel.Trace);
```
Comment (eduard-dumitru) [inline UiPath.FreeRdpClient/UiPath.FreeRdpClient.Tests/TestInfra/TestHost.cs:99]: 👍

**Review by eduard-dumitru — CHANGES_REQUESTED** [review]
I would approve if it weren't for the explicit **Cdecl** dilemma in `NativeInterface.cs`, which I suspect was forgotten from older experiments.

Even though it might work, it introduces a surprise factor in the unlikely event that we would augment the parameters and flow outside the registry range.

File: UiPath.FreeRdpClient/UiPath.FreeRdpClient/Internals/Logging.cs [hunk 2]
```diff
@@ -66,24 +69,36 @@ private void RegisterThreadScope(string scope)
...
-    Task IHostedService.StopAsync(CancellationToken cancellationToken)
+    private void DisableNativeLogsForwarding()
     {
         LoggerFactory = null;
-        return Task.CompletedTask;
+        NativeInterface.InitializeLogging(logCallback: null, registerThreadScopeCallback: null, forwardFreeRdpLogs: false);
+        _logsForwardingEnabled = false;
+    }
+
+    internal void EnsureNativeLogsForwarding()
```
Comment (danutboanta) [inline UiPath.FreeRdpClient/UiPath.FreeRdpClient/Internals/Logging.cs:94]: As time the EnableNativeLogsForwarding is done in constructor, EnsureNativeLogsForwarding is not needed (also _logsForwardingEnabled not needed)

**Comment by mihainradu** [comment]
> I would approve if it weren't for the explicit **Cdecl** dilemma in `NativeInterface.cs`, which I suspect was forgotten from older experiments.
> 
> Even though it might work, it introduces a surprise factor in the unlikely event that we would augment the parameters and flow outside the registry range.

asta a fost rezolvata... dar nu mi-e clar cum sa o marchez

**Pull request merged**